aesara/tensor/math_opt.py

```diff
@@ -1799,8 +1799,10 @@ def local_mul_zero(fgraph, node):
                 value = get_scalar_constant_value(i)
             except NotScalarConstantError:
                 continue
-            # print 'MUL by value', value, node.inputs
-            if value == 0:
+            # Print 'MUL by value', value, node.inputs.
+            # Validate code to optimize operation to result in array.
+            # E.g. We do not want't to "optimize" `a * [1, 0, 2]`` to 0.
+            if value.any() is False:
```

Comment:
Maybe my preview review comment wasn't clear: I think `value == 0` is the (only) correct condition here.

Reply:
Sounds great @michaelosthege . When I ran its respective test, I get the following, [seen in this aforementioned comment.](https://github.com/pymc-devs/aesara/pull/304#discussion_r579749060)